create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java

```diff
@@ -195,6 +205,38 @@ private URI extractDOI() {
         return UriWrapper.fromUri(DOI_OPEN_URL_FORMAT).addChild(docTp.getMeta().getDoi()).getUri();
     }
 
+    private PublicationInstance<? extends Pages> generatePublicationInstance() {
+        var citationType = getCitationType();
+        return citationType
+            .map(this::convertCitationTypeToPublicationInstance)
+            .orElseThrow(() -> new UnsupportedCitationTypeException(
+                getUnsupportedCitationTypeMessage()));
+    }
+
+    private String getUnsupportedCitationTypeMessage() {
+        return String.format(UNSUPPORTED_CITATION_TYPE_MESSAGE, docTp.getMeta().getEid());
+    }
+
+    private PublicationInstance<? extends Pages> convertCitationTypeToPublicationInstance(
```

Comment:
do not return null. Return Optional and use flatMap in line 211.

Reply:
Fixed